snap/pack/pack.go

```diff
@@ -107,27 +107,33 @@ func CheckSkeleton(w io.Writer, sourceDir string) error {
 }
 
 func loadAndValidate(sourceDir string) (*snap.Info, error) {
-	// ensure we have valid content
+	// Parsing of snap info is duplicated in ReadInfoFromSnapFile. It is done
+	// here to retrieve the snap instance name, if available, to use in case of
+	// ReadInfoFromSnapFile error
 	yaml, err := ioutil.ReadFile(filepath.Join(sourceDir, "meta", "snap.yaml"))
```

Comment:
This duplication is annoying, but provides a simple way to not change any external error messages.

Reply:
yes, not entirely a fan of this

Reply:
It does not seem worth sticking to the ReadInfoFromSnapFile based approach, which means this is not a problem anymore.